runtime/vm/resolvesupport.cpp

```diff
@@ -951,11 +972,10 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 			}
 		
 			if (ramCPEntry != NULL) {
-				UDATA valueOffset = fieldOffset;
+				UDATA valueOffset = fieldOffset;				
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-				if ('Q' == *J9UTF8_DATA(signature)) {
+				if ((!fccEntryFieldSet) &&  ('Q' == J9UTF8_DATA(signature)[0])) {
```

Comment:
this can be done differently without code duplication

something like

```
if (signature.isQtype) {
     if (!fccEntryIsSet) {
             //set fcc
            //issue write barrier
    }
    // set CP entry
}

Reply:
@tajila I optimized this part to avoid code duplication and the code is slightly different from the example you provided me with. Could you please review and see if this is fine? Thanks!